pkg/deployments/manager.go

```diff
@@ -109,6 +109,16 @@ func (r *Manager) getScaler(deploymentName string) *scaler {
 	return b
 }
 
+func (r *Manager) iterateScalers(f func(deployment string, scaler *scaler) bool) {
```

Comment:
My preference would be for this function to be refactored to lock, iterate, unlock, and return a slice of metrics as opposed to being a generic mutex protected loop with a passed-in function. It made it hard for me to understand the expected lock-duration without visiting the call-site when written as-is.

Reply:
Makes sense. Good feedback!